xonsh/main.py

```diff
@@ -224,8 +224,8 @@ def main(argv=None):
         # run a script contained in a file
         path = os.path.abspath(os.path.expanduser(args.file))
         if os.path.isfile(path):
-            sys.argv = args.args
-            env['ARGS'] = [args.file] + args.args
+            sys.argv = [args.file] + args.args
+            env['ARGS'] = sys.argv[:]  # $ARGS is not sys.argv
```

Comment:
If `$ARGS is sys.argv`, a subtle bug turns up, since, strictly speaking, `$ARGS` is not an alias. Someone can easily reassign it without even noting. For example:

```py
$ARGS = $ARG0 + [...] + $ARGS[1:]
# This is not equivalent to $ARGS[1:1] = [...]!
```

Or more complicated one:

```py
def preprocess_args():
    switch = False
    unhandled = [$ARG0]
    for arg in $ARGS[1:]:
        if arg == "--switch":
            switch = True
        elif arg == "--no-switch":
            switch = False
        else:
            unhandled.append(arg)

    $ARGS = unhandled
    return { "switch": switch }

...

# Innocent code:
switches = preprocess_args()
args = parser.parse_args()
```

The very fact that `$ARGS += something` works but `$ARGS = $ARGS + something` doesn't is extremely counter-intuitive.